pkg/controllers/sync/controller.go

```diff
@@ -1186,3 +1245,135 @@ func (s *SyncController) reconcileClusterForCascadingDeletion(
 
 	return worker.StatusAllOK
 }
+
+func (s *SyncController) listManagedClusterObjects(
+	ctx context.Context,
+	ftc *fedcorev1a1.FederatedTypeConfig,
+	clusterName string,
+) ([]*unstructured.Unstructured, sets.Set[string], error) {
+	clusterObjs := []*unstructured.Unstructured{}
+	namespaces := sets.New[string]()
+
+	resourceLister, hasSynced, exists := s.fedInformerManager.GetResourceLister(ftc.GetSourceTypeGVK(), clusterName)
+	if !exists {
+		return nil, nil, fmt.Errorf("failed to get dynamic client")
+	}
+
+	// If cluster cache is synced, we check the store.
+	// Otherwise, we will have to issue a list request.
+	if hasSynced() {
+		objects, err := resourceLister.List(labels.Everything())
+		if err != nil {
+			return nil, nil, err
```

Comment:
Let's wrap the errors in this function for more informative messages.

Reply:
The error message has been wrapped.